File: codequest-api/app/schemas/ai.py
"""AI对话相关的 Pydantic 模型"""

from typing import Optional
from pydantic import BaseModel, Field


class ChatRequest(BaseModel):
    """AI聊天请求"""
    message: str = Field(..., min_length=1, description="用户消息")
    context: Optional[dict] = None  # 可包含 lesson_title, code, error


class ChatResponse(BaseModel):
    """AI聊天响应"""
    reply: str = Field(..., description="AI回复内容")

File: HelloWorld-web/src/api/ai.ts
import apiClient from './client'

export type AIMode = 'tutor' | 'diagnostic' | 'review' | 'planning'

export interface ChatMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface ChatRequest {
  message: string
  mode?: AIMode
  context?: {
    lesson_id?: number
    lesson_title?: string
    language?: string
    code?: string
    error?: string
    rag_results?: Array<{
      title?: string
      content?: string
      snippet?: string
      score?: number
    }>
  }
}

export interface ChatResponse {
  reply: string
}

/** 发送AI聊天消息（非流式） */
export async function sendChatMessage(request: ChatRequest): Promise<ChatResponse> {
  const { data } = await apiClient.post<ChatResponse>('/ai/chat', request)
  return data
}
